### PR TITLE
Add Bonus Unit 3: Advanced Topics

### PR DESCRIPTION
**Pull request by simoninithomas**
This is the Unit about Advanced Topics overviews (Decision Transformers, RLHF, MineRL environment, Offline RL)

Content: Decision Transformers, RLHF...

🗓️  **Publication date: 03/01/2023**

TODO:
:heavy_check_mark:  Finish the illustrations (thumbnail)
:heavy_check_mark: @edbeeching for Decision Transformers tutorial we would need to update the colab? There's a bug + wdyt for the text? Something to add? (the unit visualization is available below)
:heavy_check_mark: @natolambert for RLHF do I have something to add? I was wondering maybe a small introduction to RLHF. (the unit visualization is available below)
:heavy_check_mark:@natolambert (you again :laughing: ) I was wondering if you have some good model-based RL content to share with the classmates on this unit?
:heavy_check_mark: @ClementRomac wdyt of a section about your research project? The presentation you gave last time was super nice

### CODE REVIEW

File: units/en/unitbonus3/model-based.mdx [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Model Based Reinforcement Learning
+
+Nathan can you provide an introduction and good learning resources?
```
Comment (natolambert) [inline units/en/unitbonus3/model-based.mdx:3]:
```suggestion
# Model-based reinforcement learning (MBRL)

Model-based reinforcement learning only differs from it’s model-free counterpart in the learning of a *dynamics model*, but that has substantial downstream effects on how the decisions are made. 
The dynamics models most often model the environment transition dynamics, \\( s_{t+1} = f_\theta (s_t, a_t) \\), but things like inverse dynamics models (mapping from states to actions) or reward models (predicting rewards) can be used in this framework.

**Simple version**:

There is an agent that repeatedly tries to solve a problem, accumulating state and action data. 
With that data, the agent creates a structured learning tool -- a dynamics model -- to reason about the world. 
With the dynamics model, the agent decides how to act by predicting into the future. 
With those actions, the agent collects more data, improves said model, and hopefully improves future actions.

**Academic version**:
 
Model-based reinforcement learning (MBRL) follows the framework of an agent interacting in an environment, learning a model of said environment, and then leveraging the model for control. 
Specifically, the agent acts in a Markov Decision Process (MDP) governed by a transition function \\( s_{t+1} = f (s_t , a_t) \\) and returns a reward at each step \\( r(s_t, a_t) \\). With a collected dataset \\( D :={ s_i, a_i, s_{i+1}, r_i} \\), the agent learns a model, \\( s_{t+1} = f_\theta (s_t , a_t) \\) to minimize the negative log-likelihood of the transitions. 
We employ sample-based model-predictive control (MPC) using the learned dynamics model, which optimizes the expected reward over a finite, recursively predicted horizon, \\( \tau \\), from a set of actions sampled from a uniform distribution \\( U(a) \\), (see [paper](https://arxiv.org/pdf/2002.04523) or [paper](https://arxiv.org/pdf/2012.09156.pdf) or [paper](https://arxiv.org/pdf/2009.01221.pdf)).

## Further reading
For more information on MBRL, we recommend you check out the following resources.

1. A [recent review paper on MBRL (long)](https://arxiv.org/abs/2006.16712),
2. A [blog post on debugging MBRL](https://www.natolambert.com/writing/debugging-mbrl).

```

File: units/en/unitbonus3/rlhf.mdx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# RLHF
+
+- Introduction to RL HF: Nathan
+
+## Start Learning about RLHF
+
+To start learning about RLHF:
+
+1. Read [Illustrating Reinforcement Learning from Human Feedback (RLHF)](https://huggingface.co/blog/rlhf)
```
Comment (natolambert) [inline units/en/unitbonus3/rlhf.mdx:9]:
```suggestion
1. Read this introduction: [Illustrating Reinforcement Learning from Human Feedback (RLHF)](https://huggingface.co/blog/rlhf).
```

File: units/en/unitbonus3/rlhf.mdx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+2. Watch the recorded live we did some weeks ago, where Nathan covered the basics of Reinforcement Learning from Human Feedback (RLHF) and how this technology is being used to enable state-of-the-art ML tools like ChatGPT.
+Most of the talk is an overview of the interconnected ML models. It covers the basics of Natural Language Processing and RL and how RLHF is used on large language models. We then conclude with the open question in RLHF.
+
+<Youtube id="2MBJOuVq380" />
+
+3. [Closed-API vs Open-source continues: RLHF, ChatGPT, data moats](https://robotic.substack.com/p/rlhf-chatgpt-data-moats)
```
Comment (natolambert) [inline units/en/unitbonus3/rlhf.mdx:16]:
```suggestion
3. Read other blogs on this topic, such as [Closed-API vs Open-source continues: RLHF, ChatGPT, data moats](https://robotic.substack.com/p/rlhf-chatgpt-data-moats). Let us know if there are more you like!
```

File: units/en/unitbonus3/rlhf.mdx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+
+## Additional readings
+
```
Comment (natolambert) [inline units/en/unitbonus3/rlhf.mdx:20]:
```suggestion

*Note, this is copied from the Illustrating RLHF blog post above*.
```

File: units/en/unitbonus3/rlhf.mdx [hunk 5]
```diff
@@ -0,0 +1,40 @@
+# RLHF
+
+- Introduction to RL HF: Nathan
```
Comment (natolambert) [inline units/en/unitbonus3/rlhf.mdx:3]:
```suggestion
Reinforcement learning from human feedback (RLHF) is a methodology for integrating human data labels into a RL-based optimization process. 
It is motivated by the challenge of modeling human preferences. 
For many questions, even if you could try and write down an equation for one ideal, humans differ on their preferences.
Updating models based on measured data is an avenue to try and alleviate these inherently human ML problems.
```

**Review by natolambert — COMMENTED** [review]
@simoninithomas let me know how these look.

**Comment by simoninithomas** [comment]
Some updates about this unit:
- @ClementRomac do you have some resources for your research project? The presentation you gave last time was super nice. You have a preview of the unit here: https://moon-ci-docs.huggingface.co/deep-rl-course/pr_177
- @natolambert you mentioned the paper about documentation do you want to add it to this bonus unit?
- @edbeeching we will need to check when you come back how to update decision transformer training bug for the tutorial :smile:

**Comment by natolambert** [comment]
@simoninithomas Sure, when do you need the content for documentation? It shouldn't take me & my friend Tom long.

**Comment by simoninithomas** [comment]
@natolambert we're suppose to publish this unit in 3 week but the sooner the better :smile: for instance next week. I mean we can just add a small section with the link to the docs.

Thanks again by the way for you nice content on model based RL :hugs:

File: units/en/unitbonus3/rl-documentation.mdx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+and the resonances between deployed supervised machine learning systems and the sequential decision-making utilized in RL.
+
+At a minimum, Reward Reports are an opportunity for RL practitioners to deliberate on these questions and begin the work of deciding how to resolve them in practice.
+​
+## Capturing temporal behavior with documentation
+
+The core piece specific to documentation designed for RL and feedback-driven ML systems is a *change-log*. The change-log updates information
+from the designer (changed training parameters, data, etc.) along with noticed changes from the user (harmful behavior, unexpected responses, etc.).
+
+The change-log is accompanied by update triggers that encourage monitoring of these effects.
```
Comment (osanseviero) [inline units/en/unitbonus3/rl-documentation.mdx:41]:
```suggestion
The change log is accompanied by update triggers that encourage monitoring these effects.
```

**Review by osanseviero — APPROVED** [review]
Very cool! All content is easily understandable!

File: units/en/unitbonus3/decision-transformers.mdx [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# Decision Transformers
+
+The Decision Transformer model was introduced by ["Decision Transformer: Reinforcement Learning via Sequence Modeling” by Chen L. et al](https://arxiv.org/abs/2106.01345). It abstracts Reinforcement Learning as a conditional-sequence modeling problem.
```
[the remark this answers is not in the document]
Comment (simoninithomas) [reply units/en/unitbonus3/decision-transformers.mdx:3]: ✔️

**Pull request merged**